File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/CanReviewResponse.java
package com.cinema.ticket_booking.data.model.response;

import com.google.gson.annotations.SerializedName;

public class CanReviewResponse {
    @SerializedName("canReview")
    private boolean canReview;
    
    @SerializedName("alreadyReviewed")
    private boolean alreadyReviewed;
    
    @SerializedName("bookingId")
    private String bookingId;
    
    @SerializedName("existingReview")
    private ReviewResponse existingReview;

    public boolean isCanReview() {
        return canReview;
    }

    public boolean isAlreadyReviewed() {
        return alreadyReviewed;
    }

    public String getBookingId() {
        return bookingId;
    }

    public ReviewResponse getExistingReview() {
        return existingReview;
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/CanReviewResponse.kt
package com.cinema.ticket_booking.data.model.response

import com.google.gson.annotations.SerializedName

data class CanReviewResponse(
    @SerializedName("canReview")
    @get:JvmName("isCanReview")
    val canReview: Boolean,

    @SerializedName("alreadyReviewed")
    @get:JvmName("isAlreadyReviewed")
    val alreadyReviewed: Boolean,

    @SerializedName("bookingId")
    val bookingId: String?,

    @SerializedName("existingReview")
    val existingReview: ReviewResponse?
)
